plugins/plugin-chart-table/src/TableChart.tsx

```diff
@@ -17,34 +17,73 @@
  * under the License.
  */
 import React, { useState } from 'react';
+import { ColumnInstance, Column } from 'react-table';
+import {
+  FaSort,
+  FaSortUp, // asc
+  FaSortDown, // desc
+} from 'react-icons/fa';
+
+import { t } from '@superset-ui/translation';
 import { DataRecordValue, DataRecord } from '@superset-ui/chart';
-import { filterXSS } from 'xss';
-import { TableChartTransformedProps, DataColumnMeta, DataType } from './types';
-import Table, { UseColumnCellPropsColumnOption } from './DataTable';
 
-function isProbablyHTML(text: string) {
-  return /<[^>]+>/.test(text);
+import { TableChartTransformedProps, DataType } from './types';
+import DataTable from './DataTable';
+import Styles from './Styles';
+import formatValue from './utils/formatValue';
+import { PAGE_SIZE_OPTIONS } from './controlPanel';
+
+function SortIcon({ column }: { column: ColumnInstance }) {
+  const { isSorted, isSortedDesc } = column;
+  let sortIcon = <FaSort />;
+  if (isSorted) {
+    sortIcon = isSortedDesc ? <FaSortDown /> : <FaSortUp />;
+  }
+  return sortIcon;
 }
 
 /**
- * Format text for cell value
+ * Cell background to render columns as horizontal bar chart
  */
-function formatValue({ formatter }: DataColumnMeta, value: DataRecordValue): [boolean, string] {
-  if (value === null) {
-    return [false, 'N/A'];
-  }
-  if (formatter) {
-    // in case percent metric can specify percent format in the future
-    return [false, formatter(value as number)];
-  }
-  if (typeof value === 'string') {
-    const htmlText = filterXSS(value, { stripIgnoreTag: true });
-    return isProbablyHTML(htmlText) ? [true, htmlText] : [false, value];
+function cellBar({
```

Comment:
Nothing changes for this function other than moving it out of the TableChart component scope.